portal-impl/src/com/liferay/portal/service/persistence/impl/OrganizationFinderImpl.java

```diff
@@ -576,6 +591,94 @@ protected int countByOrganizationId(
 		return 0;
 	}
 
+	protected List<Object> doFindO_U_ByC_P(
+		long companyId, long parentOrganizationId,
+		QueryDefinition<?> queryDefinition) {
+
+		Session session = null;
+
+		try {
+			session = openSession();
+
+			StringBundler sb = new StringBundler(5);
+
+			sb.append(StringPool.OPEN_PARENTHESIS);
+
+			String sql = CustomSQLUtil.get(FIND_BY_C_P);
+
+			sql = StringUtil.replace(
+				sql, "SELECT organizationId",
+				"SELECT organizationId, 0 AS userId, name");
+			sql = StringUtil.replace(
+				sql, "(organizationId > ?) AND", StringPool.BLANK);
+
+			sb.append(sql);
+			sb.append(") UNION ALL (");
+
+			sql = CustomSQLUtil.get(FIND_U_BY_C_S_O);
+
+			int status = queryDefinition.getStatus();
+
+			if (status == WorkflowConstants.STATUS_ANY) {
+				sql = StringUtil.replace(
+					sql, "(User_.status = ?) AND", StringPool.BLANK);
+			}
+
+			sb.append(sql);
+			sb.append(StringPool.CLOSE_PARENTHESIS);
+
```

Comment:
The queryDefinition param also includes an orderByComparator which is not being honored by this method. I've seen in JournalFolderFinderImpl, that it's usually done right before creating teh SQLQuery by doing the following:

```
sql = CustomSQLUtil.replaceOrderBy(sql, queryDefinition.getOrderByComparator());
```